domain/db_test.go

```diff
@@ -76,8 +77,8 @@ func TestNormalSessionPool(t *testing.T) {
 	info, err1 := infosync.GlobalInfoSyncerInit(context.Background(), "t", func() uint64 { return 1 }, nil, true)
 	require.NoError(t, err1)
 	conf := config.GetGlobalConfig()
-	conf.Port = 0
-	conf.Status.StatusPort = 10045
+	conf.Port = uint(4000 + rand.Intn(1000))
```

Comment:
Please use `freeport.GetFreePorts()` instead.

Reply:
I suggest other fixes, just use `0` since it will allocate a freeport automatically. @hawkingrei 

The real problem is that, socket address is not computed using the allocated free port but 0. It can be fixed by recomputing the address here. Note L254:

https://github.com/pingcap/tidb/blob/756a81701430e3ac496a70dbcc1743b2cbe312dd/server/server.go#L254-L259

Also note that socket path computing is happend in the main binary only:

```
tidb-server/main.go
444:            cfg.Socket = *socket
657:    cfg.Socket = strings.Replace(cfg.Socket, "{Port}", fmt.Sprintf("%d", cfg.Port), 1)
```

That means the path will not changed in tests even if the port is unique.